kotlinpoet/src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -98,8 +98,9 @@ public class FunSpec private constructor(
     codeWriter.emitWhereBlock(typeVariables)
 
     val isEmptyConstructor = isConstructor && body.isEmpty()
-    if (modifiers.containsAnyOf(ABSTRACT, EXTERNAL, EXPECT) || EXPECT in implicitModifiers ||
-      isEmptyConstructor
+    val isExternal = modifiers.containsAnyOf(EXTERNAL) || implicitModifiers.containsAnyOf(EXTERNAL)
+    if (modifiers.containsAnyOf(ABSTRACT, EXPECT) || implicitModifiers.containsAnyOf(EXPECT) ||
+      isEmptyConstructor || (isExternal && body.isEmpty())
```

Comment:
nit: since we're only checking for a single entry, containsAnyOf is redundant. It was introduced to avoid allocating collections to check against multiple entries.

```suggestion
    val isExternal = EXTERNAL in modifiers || EXTERNAL in implicitModifiers
    if (modifiers.containsAnyOf(ABSTRACT, EXPECT) || EXPECT in implicitModifiers ||
      isEmptyConstructor || (isExternal && body.isEmpty())
```